libraries/classes/Navigation/NavigationTree.php

```diff
@@ -1143,9 +1143,14 @@ private function renderNode(Node $node, string $class = ''): string
         $controlButtons = '';
         $paths = $node->getPaths();
         $nodeIsContainer = $node->type === Node::CONTAINER;
-        $hasSiblingsOrIsNotRoot = $node->hasSiblings() || $node->realParent() === false;
+        $hasSiblingsOrIsNotRoot = $node->hasSiblings() || count($node->parents(false, true));
         $liClasses = '';
 
+        // Don't show the 'Tables' node under each database unless necessary
+        if ($node instanceof NodeTableContainer && ! $node->hasSiblings()) {
+            $hasSiblingsOrIsNotRoot = false;
```

Comment:
```suggestion
            // It was not siblings and is a table container so it is self contained
            $hasSiblingsOrIsNotRoot = false;
```
Normally I am not too much of a commenting code person but in this class it is more than necessary for people that will fix issues with it in 5-10 years

I commented about how I understand this if ;)